src/Transport/Sending/IBatcher.cs

```diff
@@ -1,10 +1,11 @@
 namespace NServiceBus.Transport.AzureServiceBus
 {
     using System.Collections.Generic;
+    using System.Threading.Tasks;
     using Transport;
 
     interface IBatcherInternal
     {
-        IList<BatchInternal> ToBatches(TransportOperations operations);
+        Task<List<BatchInternal>> ToBatches(TransportOperations operations);
```

Comment:
Had to do it this way as it has to run after we've gone through installers and manage right checks.
I'll check if there's an option.